### PR TITLE
[#166] Replace nat views with void entrypoints

### LINKED ISSUE #166: Remove unnecessary nat views
# Clarification and motivation

We currently have two `nat` views that are not necessary: `get_total_supply` and `get_vote_permit_counter`.
Both these views main use is for testing, which could be replaced.

More importantly these pose a potential security risk in case a `DAO` contract has the administrator role in another `DAO` or it's administrator of itself, in which voting period and quorum threshold can be changed by anyone.

These could have been just removed if #31 was ready, but that's not the case.
So, to update our tests accordingly for the moment we can (from best to worst):
1. avoid to check those values directly altogether (but keep the test indirectly as we do for other ones).
2. move the tests that depend on these to the integrational engine only.
3. replace them with `void`s and use those instead.

# Acceptance criteria

1. Tests are updated depending on which one of the solutions above is possible
2. Both on-chain views,  `get_total_supply` and `get_vote_permit_counter` are removed from the contract

Note: this issue affects both the Lorentz and Ligo version of the contract.

### PR DESCRIPTION
**Pull request by kkirka**
## Description

**Problem:** We need to somehow get data from storage but using views is insecure: other contracts can trick DAO into sending operations to arbitrary contracts.

**Solution:** Replace views with `void` entrypoints.

## Related issue(s)

Resolves #166 

#### Related changes (conditional)

- Tests
  - [x] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

[//]: # (Add more docs here if you have them in the repository)
- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

- [x] My commits comply with [the following policy](https://www.notion.so/serokell/Commit-and-PR-policy-4cf98e1a910a415d86b5f2491d9af1af).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: ligo/src/base_DAO.mligo [hunk 1]
```diff
@@ -33,7 +33,7 @@ let requiring_no_xtz (param, store, config : forbid_xtz_params * storage * confi
     | Flush (p) -> flush (p, config, store)
     | Burn (p) -> burn(p, store)
     | Mint (p) -> mint(p, store)
-    | GetVotePermitCounter (p) -> get_vote_permit_counter(p, store)
+    | Get_vote_permit_counter (p) -> get_vote_permit_counter(p, store)
```
Comment (pasqu4le) [inline ligo/src/base_DAO.mligo:36]: Good move :+1:

File: ligo/haskell/test/Test/Ligo/BaseDAO/Proposal.hs [hunk 2]
```diff
@@ -160,14 +160,13 @@ test_BaseDAO_Proposal =
           checkTokenBalance unfrozenTokenId dao proposer 48
 
           -- Check total supply
-          let getTotalSupply tokenId consumer = call dao (Call @"Get_total_supply") (mkView tokenId consumer)
-          consumer <- originateSimple "consumer" [] contractConsumer
-          withSender (AddressResolved proposer) $ getTotalSupply unfrozenTokenId consumer
-          checkStorage (AddressResolved $ toAddress consumer) (toVal [148 :: Natural]) -- initial = 200
+          withSender (AddressResolved proposer) $
+            call dao (Call @"Get_total_supply") (mkVoid unfrozenTokenId)
+              & expectError (VoidResult (148 :: Natural)) -- initial = 200
```
Comment (pasqu4le) [inline ligo/haskell/test/Test/Ligo/BaseDAO/Proposal.hs:165]: I wonder why your removed the `-- initial = *` comments from everywhere but here :thinking:

**Review by pasqu4le — COMMENTED** [review]
Only a question and a nitpick: in both the PR description and the commit message there's a `usign` typo.

Otherwise LGTM, but you'll also need to run `./scripts/generate-cabal-files.sh` to update the cabal files.

File: ligo/haskell/test/Test/Ligo/BaseDAO/Proposal.hs (hunk 2 above)
Comment (kkirka) [reply ligo/haskell/test/Test/Ligo/BaseDAO/Proposal.hs:165]: Because copypasting is hard! Reverted :)

**Review by pasqu4le — APPROVED** [review]
LGTM now :ok_hand: 

Don't forget to squash before merging

**Pull request merged**